### ISSUE TITLE
refs and resolve don't work with sharded directories

### ISSUE THREAD

**Issue by lidel** [ipfs/js-ipfs#2389]
> Filling this so we don't forget to match recent fixes in go-ipfs (https://github.com/ipfs/go-ipfs/issues/6596)

#### Version information:

0.37.0

#### Description:

`jsipfs refs` does not work with HAMT-sharded unixfs directories:

```console
$ jsipfs refs -r /ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html
Failed to get refs for /ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html: Object has no property 'wiki'
```

same problem with `ipfs resolve`:

```console
$ jsipfs resolve -r /ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/
Object has no property 'wiki'
```

#### Interop notes

`resolve` works fine in go-ipfs v0.4.22 and `refs` was fixed in https://github.com/ipfs/go-ipfs/issues/6596 and will land in v0.5.0

**Comment by achingbrain** [comment]
I wonder if we couldn't refactor all the refs code to use the unixfs-exporter - it operates on paths, will return `dag-pb`, `dag-cbor`, etc nodes and handles all the hamt stuff transparently.

**Comment by achingbrain** [comment]
Yes - it's all async iterators now and has [resolvers for common types](https://github.com/ipfs/js-ipfs-unixfs-exporter/tree/master/src/resolvers) - we can always add more.

**Comment by dirkmc** [comment]
Great 🎉

**Comment by whizzzkid** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can follow the migration plan here https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).


This issue has been resolved in Helia! if this does not address your concern please let us know  by reopening this issue before 2023-06-05!
